tests/test_operator_complex.cpp

```diff
@@ -0,0 +1,284 @@
+                    T init_im2) {                                              \
+      bool pass = true;                                                        \
+                                                                               \
+      auto std_in1 = init_std(init_re1, init_im1);                             \
+      auto std_in2 = init_std(init_re2, init_im2);                             \
+      sycl::ext::cplx::complex<T> cplx_input1{init_re1, init_im1};             \
+      sycl::ext::cplx::complex<T> cplx_input2{init_re2, init_im2};             \
+                                                                               \
+      std::complex<T> std_out{};                                               \
+      auto *cplx_out = sycl::malloc_shared<sycl::ext::cplx::complex<T>>(1, Q); \
+                                                                               \
+      std_out = std_in1 op std_in2;                                            \
+                                                                               \
+      Q.single_task([=]() {                                                    \
+         cplx_out[0] = cplx_input1 op cplx_input2;                             \
+       }).wait();                                                              \
+                                                                               \
+      pass &= check_results(cplx_out[0], std_out, /*is_device*/ true);         \
+                                                                               \
+      cplx_out[0] = cplx_input1 op cplx_input2;                                \
+                                                                               \
+      pass &= check_results(cplx_out[0], std_out, /*is_device*/ false);        \
+                                                                               \
+      return pass;                                                             \
+    }                                                                          \
+  };
+
+test_op(test_add, +);
+test_op(test_sub, -);
+test_op(test_mul, *);
+test_op(test_div, /);
+
+#undef test_op
+
+#define test_op_assign(name, op_assign)                                        \
+  template <typename T> struct name {                                          \
+    bool operator()(sycl::queue &Q, T init_re1, T init_im1, T init_re2,        \
+                    T init_im2) {                                              \
+      bool pass = true;                                                        \
+                                                                               \
+      auto std_in = init_std(init_re1, init_im1);                              \
+      sycl::ext::cplx::complex<T> cplx_input{init_re1, init_im1};              \
+                                                                               \
+      auto std_out = init_std(init_re2, init_im2);                             \
```

Comment:
do you mind renaming it to `std_inout`?  
It fills weird to initialize a `std_out` value.

Reply:
Done